src/lib/listData.ts

```diff
@@ -9,45 +9,62 @@ import {
 
 import { dHondt } from "./worksCouncils";
 
+/**
+ * tally & distribute the outcomes on per-list, and validate for
+ * works constitution act legislated gender quota distribution
+ */
 function tallyAndValidateList(
   list: ListItem,
   listId: string,
   binaryWorkplaceGenderTally: Tally,
   seatDistribution: Tally,
   overflowDistribution: number,
+  genderDistribution: number,
   minorityGender?: GenderEnum
 ): ListDataItem {
-  const listDistribution = seatDistribution && seatDistribution[listId];
+  let popularListDistribution = seatDistribution && seatDistribution[listId];
   const listGenderRatio =
     binaryWorkplaceGenderTally &&
-    listDistribution &&
-    dHondt(binaryWorkplaceGenderTally, listDistribution);
+    popularListDistribution &&
+    dHondt(binaryWorkplaceGenderTally, popularListDistribution);
 
   let isGenderRatioValid = null;
 
   let popularlyElectedMembers: number[] = [];
   let minorityGenderElectedMembers: number[] = [];
   let overflowElectedMembers: number[] = [];
+  let genderOverflowElectedMembers: number[] = [];
 
-  if (listDistribution > list.members.length) {
-    overflowDistribution += listDistribution - list.members.length;
+  if (popularListDistribution > list.members.length) {
+    overflowDistribution += popularListDistribution - list.members.length;
   }
 
   list.members.forEach((member, i) => {
-    if (i < listDistribution) {
+    if (i < popularListDistribution) {
       popularlyElectedMembers.push(i);
       if (member.gender === minorityGender) {
         minorityGenderElectedMembers.push(i);
       }
     } else {
-      // if there are overflow votes from other lists to distribute,
-      // and this list has at least one vote
-      if (overflowDistribution > 0 && list.votes > 0) {
-        overflowElectedMembers.push(i);
-        if (member.gender === minorityGender) {
-          minorityGenderElectedMembers.push(i);
+      // a list must have at least one vote for re-distribution, or is there more nuance her?
+      if (list.votes > 0) {
+        // if there are overflow votes from other lists to distribute
+        if (overflowDistribution > 0) {
+          overflowElectedMembers.push(i);
+          if (member.gender === minorityGender) {
+            minorityGenderElectedMembers.push(i);
+          }
+          overflowDistribution -= 1;
+        }
+        // or gender quota seats to distribute
+
+        if (genderDistribution > 0) {
+          if (member.gender === minorityGender) {
+            genderOverflowElectedMembers.push(i);
+            minorityGenderElectedMembers.push(i);
```

Comment:
I think here we should be careful, for example where a second list gets an overflow seat...that overflow seat still might not satisfy gender quorum, but I think the gender correction can happen regardless of whether it's an overflow, or general correction for popularly elected candidates, so this feels like a smell

Reply:
I will add some more comments to explain